runtime/include/chpl-mem.h

```diff
@@ -169,11 +164,13 @@ void* chpl_mem_array_alloc(size_t nmemb, size_t eltSize, c_sublocid_t subloc,
     if (p == NULL) {
       p = chpl_mem_allocMany(nmemb, eltSize, CHPL_RT_MD_ARRAY_ELEMENTS,
                              lineno, filename);
-      do_localize = !chpl_mem_localizes();
+      // TODO is this right?
+      do_localize = (subloc == c_sublocid_all) ? true : false;
     }
 
     if (do_localize) {
       if (subloc == c_sublocid_all) {
+        // TODO can this be removed? I think it's multi-ddata only?
```

Comment:
Yes, subloc==c_sublocid_all can only be true when we have multi-ddata, so this branch can go.

Reply:
Cool, thanks

If it's ok with you, I plan to leave all the current functions in chpl-topo for now, and I'll let you remove any you don't think we'll need

Reply:
Yes, I'd prefer we leave chpl-topo alone for now.